File: python/fibonacci_sequence/src/fibonacci.py
__all__ = ['fibonacci']

def fibonacci(init_num, iter):
    current_num = init_num
    next_num    = current_num + 1
    result      = list()
    for _ in range(init_num, iter + init_num):
        result.append(current_num)
        current_num, next_num = next_num, current_num + next_num
    return result

File: ruby/fibonacci_sequence/src/fibonacci.rb
# rbs_inline: enabled

# @rbs init_num: Integer
# @rbs iter: Integer
# @rbs return: Array[Integer]
def fibonacci(init_num:, iter:)
  current_num = init_num
  next_num    = current_num + 1
  result      = Array.new
  iter.times do
    result.push(current_num)
    current_num, next_num = next_num, current_num + next_num
  end
  result
end
